// react-native-media-clipboard/lib/commonjs/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "Clipboard", {
  enumerable: true,
  get: function get() {
    return _MediaClipboardContext.Clipboard;
  }
});
Object.defineProperty(exports, "ClipboardContext", {
  enumerable: true,
  get: function get() {
    return _MediaClipboardContext.ClipboardContext;
  }
});
Object.defineProperty(exports, "ClipboardProvider", {
  enumerable: true,
  get: function get() {
    return _MediaClipboardContext.ClipboardProvider;
  }
});
Object.defineProperty(exports, "MediaSource", {
  enumerable: true,
  get: function get() {
    return _MediaClipboard.MediaSource;
  }
});
Object.defineProperty(exports, "getClipboardMediaSource", {
  enumerable: true,
  get: function get() {
    return _MediaClipboard.getClipboardMediaSource;
  }
});
Object.defineProperty(exports, "listenToClipboardChanges", {
  enumerable: true,
  get: function get() {
    return _MediaClipboard.listenToClipboardChanges;
  }
});
Object.defineProperty(exports, "stopListeningToClipboardChanges", {
  enumerable: true,
  get: function get() {
    return _MediaClipboard.stopListeningToClipboardChanges;
  }
});
Object.defineProperty(exports, "listenToClipboardRemove", {
  enumerable: true,
  get: function get() {
    return _MediaClipboard.listenToClipboardRemove;
  }
});
Object.defineProperty(exports, "stopListeningToClipboardRemove", {
  enumerable: true,
  get: function get() {
    return _MediaClipboard.stopListeningToClipboardRemove;
  }
});
Object.defineProperty(exports, "getClipboardContents", {
  enumerable: true,
  get: function get() {
    return _MediaClipboard.getClipboardContents;
  }
});
Object.defineProperty(exports, "ClipboardResponse", {
  enumerable: true,
  get: function get() {
    return _MediaClipboard.ClipboardResponse;
  }
});

var _MediaClipboardContext = require("./MediaClipboardContext");

var _MediaClipboard = require("./MediaClipboard");
